app/models/user.rb

```diff
@@ -39,6 +39,11 @@ def github_client_scopes
     GitHubUser.new(github_client, uid).client_scopes
   end
 
+  # for searching, disable pagination so we can control the number of suggestions
+  def github_search_client
```

Comment:
Actually we can access the `token` attribute outside the class, so we can create the client in the controller.

My concerns were about concurrency, but I forgot that HTTP is stateless :sweat_smile:

> Alternatively, what about adding an options hash argument to User#github_client that would allow you to disable pagination with that method?

I thinks this is more reasonable :+1: